enhancements/cluster-logging/hypershift-audit-logs.md

```diff
@@ -0,0 +1,219 @@
+
+### User Stories
+
+#### As a cluster owner, I want to store and/or forward selected API audit logs to my chosen destinations.
+
+- configure API audit policy and forwarding using guest cluster resource.
+- have the desired audit logs forwarded from the management cluster to my destination.
+
+#### As a Hypershift SRE, I want to control plane audit logs to go to my chosen destination with my policy.
+
+- configure the SD internal audit policy
+- forward logs to the SD internal destination (splunk)
+- independent of user configuration.
+
+### Goals
+
+- Multiple separate audit policies. 
+  See [api-audit-log-policy](/api-audit-log-policy.md) for how the CLO will support this.
+- Control log forwarding from the _management_ cluster with configuration in the _guest_ cluster.
+- Keep control of all management cluster forwarding on the management cluster.
+  - guest configuration requests forwarding, management cluster controls what is delivered.
+
+### Non-Goals
+
+- Only API-server audit logs, no application, infrastructure or node audit logs.
+- May not support every possible log destination or configuration.
+  There may be restrictions imposed by the management cluster.
+
+## Proposal
+
+### Workflow Description
+
+#### Guest Cluster
+
+1. Create a `Policy.audit.k8s.io` resource with the desired policy
+   (see [api-audit-log-policy](/api-audit-log-policy.md))
+2. Create a `HypershiftLogForwarder` resource with an `audit` input linked to the policy. \
+   The `HypershiftLogForwarder` API is a restricted subset of `ClusterLogForwarder`
+3. Create `outputs` and `pipelines` as normal to forward the resulting logs.
+
+#### Management Cluster
+
+1. Deploy the existing CLO
+1. Deploy the new _hypershift logging operator_ (HLO)
+
+The _hypershift logging operator_ watches for `HypershiftLogForwarder` and related `Policy.audit.k8s.io.` resources
+in the _guest_ clusters (via the HCP API servers)
+It reconciles by generating a `ClusterLogForwarder` in the HCP namespace.
+
+The HLO only generates configuration.
+The CLO in the management cluster does the log collection and forwarding.
+
+### API Extensions
+
+#### HypershiftLogForwarder
+
+The `HyperShiftLogForwarder` resource is created on the _guest_ cluster, but is reconciled by
+the _hypershift logging operator_ on the _management_ cluster.
+
+This allows the user to specify audit forwarding configuration, but the management
+cluster controls if/how it is implemented in the HCP.
+
+`HyperShiftLogForwarder` is a subset of the `ClusterLogForwarder` API:
+- Only the `audit` input is allowed, and only API server audit logs will be forwarded.
+- There may be restrictions on the set of outputs supported from the management cluster.
+  - Possibly only a predefined set of named outputs will be allowed.
+- There may be restrictions on pipeline features that are not relevant to API audit logs.
+
+### Implementation Details
+
+#### ClusterLogForwarder multiple instances
```

Comment:
The functionality needed on the MC is captured in the `HypershiftLogForwarder`.  Could that CR be used in each HCP NS to aggregate everything?

1. SD wites a HLF to the HCP NS with the audit forwarding needed to splunk.  
2. Customer writes HLF in the hosted cluster with the customer forwarding needs.
3. HLO watches the hosted cluster HLF and adds to the HLF already created, in the status.  If HLF doesn't exist, one is created with no spec.

Something like this would mean CLF isn't changed, no breaking API changes to deal with.  All the configuration ends up in one place.  It's local to the HCP, so authentication (STS) is fine.  No need for transformation, the CRDs will be the same in the management cluster and hosted cluster.